### ISSUE TITLE
Add TITLE parameter to IFRAME of article galley view

### ISSUE THREAD

**Issue by israelcefrin** [pkp/pkp-lib#5190]
Our accessibility audit report pointed that Article galley viewing, via `iframe `element, is missing  descriptive titles parameter for a better accessibility.

We should adjust the code to add this title parameter in iframe element while include, in the descripption, an indentifier prefix for each galley, i.e.: `HTML `, `PDF`, etc...

For example, the iframe looks like this:
```
<div id="htmlContainer" class="galley_view">
<iframe name="htmlFrame" src="http://domain.org/index.php/journalname/article/download/49/41?inline=1" allowfullscreen webkitallowfullscreen></iframe>
</div>

```
This should be change to (`HTML` galley):
```
<div id="htmlContainer" class="galley_view">
<iframe name="htmlFrame" src="http://domain.org/index.php/journalname/article/download/49/41?inline=1" allowfullscreen webkitallowfullscreen title=”HTML: INDIVIDUAL JOURNAL TITLE AND WORDS HTML VERSION”></iframe>
</div>
```

Or (`PDF `galley)
```
<div id="htmlContainer" class="galley_view">
<iframe name="htmlFrame" src="http://domain.org/index.php/journalname/article/download/49/41?inline=1" allowfullscreen webkitallowfullscreen title=”PDF: INDIVIDUAL JOURNAL TITLE AND WORDS HTML VERSION”></iframe>
</div>
```

The preffix is the galley label from article landing page.

**Comment by NateWr** [comment]
You'll want to create a new locale string in `locale/en_US/submission.po` like this:

```
msgid "submission.representationOfTitle"
msgstr "{$representation} of {$title}"
```

Then use something in the galley view templates like this:

```
<iframe
  ...
  title="{translate key="submission.representationOfTitle" representation=$galley->getLabel() title=$galleyPublication->getLocalizedFullTitle()}"
>
```

Or something like that.

**Comment by asmecher** [comment]
...and off the top of my head, you'll want to use the Smarty `escape` filter to make sure HTML-ish characters in those two variables are escaped:
```
<iframe
  ...
  title="{translate key="submission.representationOfTitle" representation=$galley->getLabel()|escape title=$galleyPublication->getLocalizedFullTitle()|escape}"
>
```

**Comment by thinkbulecount2** [comment]
pkp-lib PR: https://github.com/pkp/pkp-lib/pull/5562
ojs PR: https://github.com/pkp/ojs/pull/2652
cc: @NateWr

**Comment by NateWr** [comment]
This looks good, @thinkbulecount2. This fixes for HTML galleys but not for PDF galleys. The template for those galleys is in a submodule (`plugins/generic/pdfJsViewer`) in this repository: https://github.com/pkp/pdfJsViewer.

Also, I think we'll need these changes done in OMP as well.

**Comment by thinkbulecount2** [comment]
Changes made to pdfJsViewer: https://github.com/pkp/pdfJsViewer/pull/51
cc: @NateWr

**Comment by thinkbulecount2** [comment]
omp PR: https://github.com/pkp/omp/pull/779
cc: @NateWr

**Comment by NateWr** [comment]
All merged!

**Comment by asmecher** [comment]
Hi all -- this causes a fatal error when viewing a PDF issue galley, as the `$publication` object doesn't exist for an issue galley. I've filed this separately at https://github.com/pkp/pkp-lib/issues/5762. @thinkbulecount2, do you want to take a crack at fixing it?

**Comment by asmecher** [comment]
(Actually, leaving this one closed -- work can continue at https://github.com/pkp/pkp-lib/issues/5762.)

**Comment by asmecher** [comment]
I've filed a new issue related to this in OMP that'll need to be fixed for version 3.2.1: https://github.com/pkp/pkp-lib/issues/5977
